app/mailers/placements/provider_user_mailer.rb

```diff
@@ -28,61 +28,55 @@ def user_membership_destroyed_notification(user, organisation)
 ========
 
   def partnership_created_notification(user, source_organisation, partner_organisation)
-    notify_email(
-      to: user.email,
-      subject: t(".subject", organisation: partner_organisation.name),
-      body: t(
-        ".body",
-        user_name: user.full_name,
-        source_organisation: source_organisation.name,
-        partner_organisation: partner_organisation.name,
-        link: placements_provider_partner_schools_url(partner_organisation),
-      ),
-    )
+    @user_name = user.first_name
+    @school_name = source_organisation.name
+    @provider_name = partner_organisation.name
+    @itt_contact_email = source_organisation.school_contact_email_address
+    @sign_in_url = sign_in_url
+    @service_name = service_name
+
+    notify_email to: user.email, subject: t(".subject")
   end
 
   def partnership_destroyed_notification(user, source_organisation, partner_organisation)
-    notify_email(
-      to: user.email,
-      subject: t(".subject", organisation: partner_organisation.name),
-      body: t(
-        ".body",
-        user_name: user.full_name,
-        source_organisation: source_organisation.name,
-        partner_organisation: partner_organisation.name,
-        link: placements_provider_partner_schools_url(partner_organisation),
-      ),
-    )
+    @user_name = user.first_name
+    @provider_name = partner_organisation.name
+    @school_name = source_organisation.name
+    placements_school = source_organisation.becomes(Placements::School)
```

Comment:
Is this 'becomes' call definitely necessary?

```suggestion
```

I'm surprised because I think this should already be a `Placements::School` object. At least in practice?

The call chain for this mailer seems to be:

- [app/services/placements/partnerships/notify/remove.rb](https://github.com/DFE-Digital/itt-mentor-services/blob/f599c8ce8a397eed22f89e12e0c455417e94e868/app/services/placements/partnerships/notify/remove.rb)
- [app/controllers/placements/schools/partner_providers_controller.rb](https://github.com/DFE-Digital/itt-mentor-services/blob/dcd0d7b9672b5b3561d038c75af8f107f7877e90/app/controllers/placements/schools/partner_providers_controller.rb#L6-L7)
- [app/controllers/placements/partnerships_controller.rb](https://github.com/DFE-Digital/itt-mentor-services/blob/62b143328ded910f225f629f5e1a7e301fe83c73/app/controllers/placements/partnerships_controller.rb#L14-L15)

...where [`source_organisation` is `@school`](https://github.com/DFE-Digital/itt-mentor-services/blob/dcd0d7b9672b5b3561d038c75af8f107f7877e90/app/controllers/placements/schools/partner_providers_controller.rb#L15-L21), which [should be a `Placements::School`](https://github.com/DFE-Digital/itt-mentor-services/blob/dcd0d7b9672b5b3561d038c75af8f107f7877e90/app/controllers/placements/application_controller.rb#L7-L9).

I wonder if this is just a bad spec where we're passing in the wrong type of school, but it's actually okay in the implementation? 🤔